app/controllers/tags_controller.rb

```diff
@@ -196,7 +196,7 @@ def create
       else
         flash[:notice] = ts('Tag was successfully created.')
       end
-      @tag.update_attribute(:canonical, tag_params[:canonical])
+      @tag.update_attribute(:canonical, tag_params[:canonical]) unless @tag.canonical? # If tag already canonical, do not uncanonize it
```

Comment:
I'm wondering if this makes more sense to do at the model layer, maybe something like
```ruby
before_update :prevent_un_canonicalization, if: :will_save_change_to_canonical?
def prevent_un_canonicalization
  return unless canonical_was
  self.canonical = true
end
```
(caveat: I think there might be a better way to do that, but I didn't want to go too far down the rabbit hole.)

Personally, erroring here seems more user-friendly as well (it also is kinda weird to me that tag creation could update at all...), but someone more official should probably weigh in on that question.

Reply:
I don't think putting that callback in the model is a good plan, as it'll prevent anyone from ever making a tag non-canonical through the tag edit form (even for tags where that's allowed, and for admins who are always permitted to make tags non-canonical).

Reply:
Fair enough! I think it would be possible to add an exception for that, but I agree it's murkier taking that into consideration